collector/src/main/java/com/bitmovin/analytics/retryBackend/RetrySample.kt

```diff
@@ -0,0 +1,16 @@
+package com.bitmovin.analytics.retryBackend
+
+import com.bitmovin.analytics.data.AdEventData
+import com.bitmovin.analytics.data.EventData
+import java.util.Date
+
+class RetrySample(val eventData: EventData?, val adEventData: AdEventData?, var totalTime: Int, var scheduledTime: Date, var retry: Int) : Comparable<RetrySample> {
+    override fun compareTo(other: RetrySample): Int {
```

Comment:
This is a bit misleading as `compareTo` sounds like it's actually comparing the two samples, whereas this only compares the scheduleTimes. This could lead to all kinds of issues down the road if someone doesn't bother to read the implementation of `compareTo`

Reply:
Moved this to `Comparator`